mod/quiz/classes/external/update_filter_condition.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+namespace mod_quiz\external;
+
+defined('MOODLE_INTERNAL') || die();
+
+require_once($CFG->libdir . '/externallib.php');
+require_once($CFG->dirroot . '/question/editlib.php');
+require_once($CFG->dirroot . '/mod/quiz/locallib.php');
+
+use external_function_parameters;
+use external_single_structure;
+use external_value;
+use external_api;
+
+/**
+ * Add random questions to a quiz.
+ *
+ * @package    mod_quiz
+ * @copyright  2022 Catalyst IT Australia Pty Ltd
+ * @author     Nathan Nguyen <nathannguyen@catalyst-au.net>
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+class update_filter_condition extends external_api {
+
+    /**
+     * Parameters for the web service function
+     *
+     * @return external_function_parameters
+     */
+    public static function execute_parameters(): external_function_parameters {
+        return new external_function_parameters (
+            [
+                'cmid' => new external_value(PARAM_INT, 'The cmid of the quiz'),
+                'id' => new external_value(PARAM_INT, 'The id of the question set reference'),
+                'filtercondition' => new external_value(PARAM_TEXT, 'Filter condition'),
+            ]
+        );
+    }
+
+    /**
+     *  Add random questions.
+     *
+     * @param int $cmid course module id
+     * @param int $id question set reference id
+     * @param string $filtercondition
+     * @return array  result
+     */
+    public static function execute(int $cmid, int $id, string $filtercondition): array {
+        global  $DB;
+
+        [
+            'cmid' => $cmid,
+            'id' => $id,
+            'filtercondition' => $filtercondition,
+        ] = self::validate_parameters(self::execute_parameters(), [
+            'cmid' => $cmid,
+            'id' => $id,
+            'filtercondition' => $filtercondition,
+        ]);
+
+        // Validate context.
+        $thiscontext = \context_module::instance($cmid);
+        self::validate_context($thiscontext);
+        require_capability('mod/quiz:manage', $thiscontext);
+
+        // Update filter condition.
+        $DB->set_field('question_set_references', 'filtercondition', $filtercondition, ['id' => $id]);
+
+        return ['message' => get_string('updatefilterconditon_success', 'mod_quiz')];
```

Comment:
same as above, cant see in ui

Reply:
It is used in this js: https://github.com/catalyst/moodle-MDL-72752/pull/63/files#diff-404b42454a1a8069c4d1790521c736773515d6b7cdf7fe160f8ae3d5a0dca483R46